src/citrine/resources/dataset.py

```diff
@@ -221,6 +222,58 @@ def register(self, data_concepts_resource: ResourceType) -> ResourceType:
         if isinstance(data_concepts_resource, ConditionTemplate):
             return self.condition_templates.register(data_concepts_resource)
 
+    def register_all(self, data_concepts_resources: List[ResourceType]) -> List[ResourceType]:
+        """
+        Register multiple data concepts resources to each of their appropriate collections.
+
+        Does so in an order that is guaranteed to store all linked items before the item that
```

Comment:
Why not use [taurus.util.writeable_sort_order](https://github.com/CitrineInformatics/gemd-python/blob/master/gemd/util/impl.py#L240) (note that it is still available under the taurus import; the source just moved to the `gemd` tree.  It will have the same function as below, but more DRY.  It should also make test coverage easier :-D

Reply:
I had no idea this was a thing!